lib/components/viewers/related-stops-panel.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react'
+
+import {
+  getRouteIdForPattern,
+  getStopName,
+  getStopTimesByPattern,
+  routeIsValid
+} from '../../util/viewer'
+import Icon from '../narrative/icon'
+
+import NextArrivalForPattern from './next-arrival-for-pattern'
+import RelatedPanel from './related-panel'
+import ViewStopButton from './view-stop-button'
+
+class RelatedStopsPanel extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expanded: false
+    }
+  }
+
+  _toggleExpandedView = () => {
+    this.setState({ expanded: !this.state.expanded })
+  }
+
+  render () {
+    const { expanded } = this.state
+    const {
+      childStops,
+      homeTimezone,
+      stopViewerArriving,
+      stopViewerConfig,
+      timeFormat
+    } = this.props
+    return (
+      <RelatedPanel
+        count={childStops?.length}
+        expanded={expanded}
+        onToggleExpanded={this._toggleExpandedView}
+        title='Related Stops'
+      >
+        <ul className='related-items-list list-unstyled'>
+          {childStops?.map((stop, index) => {
+            const { id } = stop
+            if (!expanded && index >= 2) return null
+            const stopTimesByPattern = getStopTimesByPattern(stop)
+            const label = getStopName(stop)
+            return (
+              <li className='related-item' key={id}>
+                <div className='stop-label'>
+                  <div className='stop-name overflow-ellipsis' title={label}>
+                    <Icon
+                      className='child-stop-icon'
+                      type={stop.routes && stop.routes[0].mode.toLowerCase()}
```

Comment:
If the routes at a stop are an empty array, a white-screen-of-death error occurs.

Also, stops can have multiple modes that stop at them, so I think a more generic stop icon should be used.

Reply:
Most stops only have one mode. If there are multiple modes per stop, we can use a generic icon, but I'd prefer to be descriptive when possible.